### PR TITLE
Shipit integration

### PR DESCRIPTION
**Pull request by GoodForOneFare**
Shipit integration ([stack here](https://shipit.shopify.io/shopify/shopify-codemod/production)).  Also tweaked CI to run lint+test via `check`.

Spent too long trying to make shipit use yarn, then realized that it's not worth it if CI is handling install/tests 🤦‍♂️

### CODE REVIEW

File: scripts/deploy.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const exec = require('child_process').exec;
+
+exec('npm publish', (error, stdout, stderr) => {
```
Comment (lemonmade) [inline scripts/deploy.js]: I was thinking - does this actually do anything than just running npm publish directly? Also, should we be yarn publishing instead?

**Review by lemonmade — COMMENTED** [review]
One question, then 👍

File: scripts/deploy.js (hunk 1 above)
Comment (GoodForOneFare) [inline scripts/deploy.js]: This script is an artifact of basing everything on the [buy button's deployment](https://github.com/Shopify/buy-button-js/blob/master/script/deploy.js).  My intent was to create a standard structure to guide the eventual shipit/npm integration.  

Now that you mention it, this does seem like overkill, so I think I'll revise my PRs to use `"deploy": "npm publish"`.

Comment (GoodForOneFare) [inline scripts/deploy.js]: We'd have to get Yarn into the shipit environment.  I was able to hack that together as part of a deployment, but it wasn't pretty (and without a cache dir, it seemed slower than npm).

My thought was to not worry about Yarn until we got to the point of changing the shipit-engine code, but I can start that process now if you think it'll speed things along?

Comment (lemonmade) [inline scripts/deploy.js]: Skipping `yarn` for now makes sense to me, as does the simpler deploy script 👍

**Comment by GoodForOneFare** [comment]
Simplified the deploy script & fixed up problems highlighted in other PRs.  Thank you for staring at all of these incredibly-similar-but-subtly-different commits :)

**Pull request merged**